### PR TITLE
Refactor shiny.fluent to use the updated shiny.react

### PR DESCRIPTION
**Pull request by kamilzyla**
This is a parallel PR to [shiny.react#7](https://github.com/Appsilon/shiny.react/pull/7).

### CODE REVIEW

File: inst/examples/Calendar.R [hunk 1]
```diff
@@ -1,23 +1,10 @@
 library(shiny.fluent)
 
 if (interactive()) {
-  dayPickerStrings <- jsonlite::fromJSON(
```
Comment (marekrogala) [inline inst/examples/Calendar.R:4]: why don't we need this anymore?

**Review by marekrogala — CHANGES_REQUESTED** [review]
Congrats! :tada: All examples work briliantly.

50936701_976950922497864_4810986498260205568_n


 I have a few comments, questions and requests below.


This is a prelimiary review, i didn't look much at the code, just did basic testing.




Adding all comments from examples testing here, but some of them may actually be related more to shiny.react itself.

1. Why rename `srcjs` to `nodejs` ? the former seems like a standard in R packages and apps developed at Appsilon
2. I think this previously was not wrapped in a list. Should it be?
Selection_625
3. I have mixed feelings regarding the `.i` suffix -- I really don't like it's ambiguity. Maybe we can have `Checkbox.shinyInput`? more keystrokes, but at least no magic. It will be more clear to people what's the difference between `Checkbox` and `Checkbox.shinyInput` than between `Checkbox` and `Checkbox.i`. Also, this may be especially confusing with buttons -- what should be do about them? `.shinyButton`?
4. In the components documentation dashboard (`examples/dashboard`) looks like the documentation is no longer picked up 
image
 5. I think we need examples of the new style of using components (like how do I use `Checkbox` not `Checkbox.i`, `Calendar` not `Calendar.i` etc), right?
 6. I found 1 thing that was not working: `examples/tutorial/dashboard_layout.R` fails due to a missing function `shiny.react::enable_react_debug_mode()`. Looking at master, this function is there -- perhaps you branched off before this change was in shiny.react? or did you remove it?

**Comment by kamilzyla** [comment]
Marek, thanks a lot for your effort! :slightly_smiling_face: Answers for your questions:
1. Actually, `srcjs` -> `node/src`. I wanted to put all files related to Node into a single directory to keep things tidy. I figured `node` suits this intention better and I didn't want to have `srcjs/src`.
2. Yes, I intentionally changed the ComboBox interface. This way the text entered is always available as `$text`, and if it's one of the predefined options - you can also access `$key`. I think this interface is easier and more natural to use. Previously you had to use `is.character` to differentiate between custom text and predefined option.
3. Done. I used `.shinyInput` for buttons as well - they are also inputs (they take `inputId` and can be observed via `input$` in the server).
4. Fixed.
5. We'll definitely need additional examples. I'm thinking of a vignette explaining the difference between e.g. plain components and `.shinyInput` components. I'd like to do it in a separate PR though :wink:.
6. Fixed.

File: inst/examples/dashboard/example_page.R [hunk 2]
```diff
@@ -17,9 +17,14 @@ Rd2list <- function(Rd) {
...
 
 getHelpList <- function(...) {
-  thefile <- help(...)
-  myrd <- utils:::.getHelpFile(thefile)
-  Rd2list(myrd)
+  # thefile <- help(...)
+  # myrd <- utils:::.getHelpFile(thefile)
+  # Rd2list(myrd)
+  list(
+    description = "TODO",
```
[the remark this answers is not in the document]
Comment (kamilzyla) [reply inst/examples/dashboard/example_page.R]: Fixed.

File: inst/examples/dashboard/www/examples_scripts.js [hunk 3]
```diff
@@ -86,26 +81,29 @@ const keytipConfig = {
...
       keytipProps: keytipMap.ButtonMenuItem2,
-      onClick: () => Shiny.setInputValue('buttonk3', Math.random())
+      onClick: () => Shiny.setInputValue('button3', Math.random())
     },
   ],
 };
 
 // Marquee
+const React = modules['react'];
+const Fluent = modules['@fluentui/react'];
```
[the remark this answers is not in the document]
Comment (kamilzyla) [reply inst/examples/dashboard/www/examples_scripts.js]: Thanks :slightly_smiling_face:

File: inst/examples/Calendar.R (hunk 1 above)
Comment (kamilzyla) [reply inst/examples/Calendar.R:4]: We never needed this - if the `dayPickerStrings` prop is not provided, default settings are used.

File: node/webpack.config.babel.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  module: {
+    rules: [
+      {
+        test: /\.(js|jsx)$/,
+        use: ['babel-loader'],
+      },
+    ],
+  },
+  externals: {
+    '@/shiny.react': 'modules["@/shiny.react"]',
```
[the remark this answers is not in the document]
Comment (kamilzyla) [reply node/webpack.config.babel.js]: Instead of variously named global variables for accessing modules (`React` for `react`, `ReactDOM` for `react-dom`, `window['shiny.react']` for our own `shiny.react` code), I want to introduce the following convention:
1. NPM module `name` is available as `jsmodule['name']`,
2. Custom JS bundled with R package `name` is available as `jsmodule['@/name']`.

File: inst/examples/002_more_inputs/app.R [hunk 5]
```diff
@@ -32,84 +31,79 @@ server <- function(input, output, session) {
...
 
-  lapply(c(1:8, 10:14), function(i) {
-    inputName <- paste0("input", i)
-    outputName <- paste0("textOutput", i)
+  lapply(1:14, function(i) {
+    inputName <- paste0("value", i)
+    outputName <- paste0("text", i)
     output[[outputName]] <- renderText({
-      sprintf("Value: %s", input[[inputName]])
+      value <- capture.output(dput(input[[inputName]]))
```
[the remark this answers is not in the document]
Comment (kamilzyla) [reply inst/examples/002_more_inputs/app.R:87]: It is not needed, but `dput()` provides more useful output for this example. For value being `NULL`, `FALSE`, or `"FALSE"` we get `value: NULL`, `value: FALSE` and `value: "FALSE"` respectively - instead of empty string in the first case and `value: FALSE` in the other two cases.

Unfortunately, `dput()` can only print to file / standard output - that's what the `capture.output` is for. I extracted this into a function.

File: node/webpack.config.babel.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (kamilzyla) [reply node/webpack.config.babel.js]: I'm going to include this info in a vignette for shiny.react. Note: I changed the name from `modules` to `jsmodule` after discovering that the former would prevent custom tiles from working in Leaflet (due to a bug in Leaflet, actually).

File: install.sh [hunk 6]
```diff
@@ -2,10 +2,10 @@
 set -eu
 
 # Install the JavaScript dependencies
-yarn
+(cd node && yarn)
```
Comment (marekrogala) [inline install.sh:5]: 1. > Actually, srcjs -> node/src. I wanted to put all files related to Node into a single directory to keep things tidy. I figured node suits this intention better and I didn't want to have srcjs/src.

How about `js` and `js/src` ? I don't like `node` because this suggests there's some node application and basically node is not the important point here. This is the JS part of our codebase, not Node part of our codebase -- we're just using node to build the JS part of the frontend


2. since we're making changes to this path and it is repeated, I suggest making this a variable

File: node/src/inputs.jsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
+import * as Fluent from '@fluentui/react';
+import { ButtonAdapter, InputAdapter } from '@/shiny.react';
```
Comment (marekrogala) [inline node/src/inputs.jsx:2]: what does `from '@/shiny.react'` mean? (specifically `@/`?)

File: node/src/inputs.jsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+export const ActionButton = ButtonAdapter(Fluent.ActionButton);
+export const CommandBarButton = ButtonAdapter(Fluent.CommandBarButton);
+export const CommandButton = ButtonAdapter(Fluent.CommandButton);
+export const CompoundButton = ButtonAdapter(Fluent.CompoundButton);
+export const DefaultButton = ButtonAdapter(Fluent.DefaultButton);
+export const IconButton = ButtonAdapter(Fluent.IconButton);
+export const PrimaryButton = ButtonAdapter(Fluent.PrimaryButton);
+
+export const Calendar = InputAdapter(Fluent.Calendar, (value, setValue) => ({
```
Comment (marekrogala) [inline node/src/inputs.jsx:12]: so this is created manually, is that correct? but we're still using some of the work that the generator did, like documentation?

File: R/inputs.R [hunk 9]
```diff
@@ -0,0 +1,163 @@
+button <- function(name) {
+  function(inputId, ...) shiny.react::reactElement(
+    module = "@/shiny.fluent", name = name,
+    props = shiny.react::asProps(inputId = inputId, ...),
+    deps = shinyFluentDependency()
+  )
+}
+
+#' @rdname Button
```
Comment (marekrogala) [inline R/inputs.R:9]: how did you generate this file? anything automated or purely manually?

File: R/data.R [hunk 10]
```diff
@@ -1,11 +1,11 @@
 #' A dataset of sample people based on Fluent UI examples
 #'
 #' @source \url{https://developer.microsoft.com/en-us/fluentui#/controls/web/peoplepicker}
-"fluent_people"
+"fluentPeople"
```
Comment (marekrogala) [inline R/data.R:4]: interesting, why this change? in general the snake_case is the standard in R, that's why I made it this way. Unfortunately Shiny chose a different convention and therefore for UI components we used camelCase

File: inst/www/shiny.fluent/shiny-fluent.js.LICENSE.txt [hunk 11]
```diff
@@ -0,0 +1,14 @@
+/*! *****************************************************************************
```
Comment (marekrogala) [inline inst/www/shiny.fluent/shiny-fluent.js.LICENSE.txt:1]: hmm, is this in the right place? I mean, it looks as if this file is a license for the entire shiny.fluent, not only for the Fluent UI. It's named `shiny-fluent.js.LICENSE.txt `... Why do we need this?

**Review by marekrogala — APPROVED** [review]
Looks great! Thanks for the detailed split into commits, made the review much smoother.

File: node/src/inputs.jsx (hunk 8 above)
Comment (kamilzyla) [reply node/src/inputs.jsx:12]: Correct.

File: R/inputs.R (hunk 9 above)
Comment (kamilzyla) [reply R/inputs.R:9]: Manually. I'd like to decrease the amount of repetitive typing required, but using R's meta-programming capabilities.

File: inst/www/shiny.fluent/shiny-fluent.js.LICENSE.txt (hunk 11 above)
Comment (kamilzyla) [reply inst/www/shiny.fluent/shiny-fluent.js.LICENSE.txt:1]: Webpack generates this file along `shiny-fluent.js`, as it has Fluent UI bundled in it.

File: install.sh (hunk 6 above)
Comment (kamilzyla) [reply install.sh]: Done.
